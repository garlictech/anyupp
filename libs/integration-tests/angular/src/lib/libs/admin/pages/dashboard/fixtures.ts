import { ProductComponentSetType, ProductType } from '@bgap/domain';

export const printableOrder = {
  id: 'seeded_order_id_6',
  createdAt: '2022-01-31T15:04:37.521Z',
  userId: 'seeded_user_id_6',
  unitId: 'seeded_unit_c1_g1_1_id',
  orderNum: '000001',
  items: [
    {
      productId: 'seeded_chain_product_c1_1_id',
      variantId: 'seeded_chain_product_variant_c1_p1_1_id',
      created: null,
      productName: {
        en: 'Hamburger',
        de: 'Hamburger',
        hu: 'Hamburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: ['egg', 'gluten', 'mustard', 'milk', 'soya', 'fish', 'sesame'],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
    {
      productId: 'seeded_chain_product_c1_2_id',
      variantId: '16853de0-528d-11ec-90c0-fdfe82b3802d',
      created: null,
      productName: {
        en: 'Fishburger',
        de: 'Fishburger',
        hu: 'Halburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: [
        'lupin',
        'molluscs',
        'crustaceans',
        'egg',
        'soya',
        'sesame',
        'fish',
        'gluten',
      ],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
    {
      productId: 'seeded_chain_product_c1_3_id',
      variantId: 'd22dee40-528b-11ec-afe3-15e1afd659e2',
      created: null,
      productName: {
        en: 'Sajtburger',
        de: 'Sajtburger',
        hu: 'Sajtburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: ['sesame'],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
  ],
  paymentMode: {
    type: 'cash',
    caption: null,
    method: 'cash',
  },
  statusLog: [
    {
      userId: 'test-alice',
      status: 'placed',
      ts: 1627909024677,
    },
  ],
  archived: false,
  takeAway: false,
  place: {
    seat: '02',
    table: '01',
  },
  paymentIntention: null,
  transactionStatus: 'success',
  transactionId: 'seeded_transaction_id_6',
  unpayCategory: null,
  orderMode: 'instant',
  servingMode: 'inPlace',
  serviceFee: {
    currency: 'HUF',
    netPrice: 200,
    taxPercentage: 27,
  },
  packagingSum: null,
  sumPriceShown: {
    currency: 'HUF',
    pricePerUnit: 298.2,
    priceSum: 2982,
    tax: 27,
    taxSum: 633.96,
  },
  rating: {
    key: 'RATING KEY',
    value: 3,
  },
  hasRated: true,
  tip: null,
  tipTransactionStatus: null,
  tipTransactionId: null,
  updatedAt: '2022-01-31T15:04:37.521Z',
  transaction: {
    id: 'seeded_transaction_id_6',
    userId: 'test-alice',
    orderId: 'seeded_order_id_6',
    type: 'cash',
    total: 0,
    currency: 'HUF',
    status: 'waiting_for_payment',
    externalTransactionId: null,
    invoiceId: null,
    receiptId: null,
    createdAt: '2022-01-31T15:04:37.267Z',
    updatedAt: '2022-01-31T15:04:37.267Z',
    paymentMethodId: null,
    user: {
      id: 'test-alice',
      name: 'Mekk Elek',
      email: 'testuser+alice@anyupp.com',
      phone: '1234',
      profileImage: null,
      stripeCustomerId: null,
      invoiceAddress: null,
      createdAt: '2022-01-31T15:04:19.739Z',
      updatedAt: '2022-01-31T15:04:19.739Z',
    },
    invoice: null,
    receipt: null,
  },
  tipTransaction: null,
};
