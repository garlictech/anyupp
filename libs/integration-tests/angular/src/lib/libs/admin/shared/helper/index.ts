import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
