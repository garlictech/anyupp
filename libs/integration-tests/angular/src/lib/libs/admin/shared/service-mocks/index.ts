export class MockNbDialogService {
  open() {
    /** */
  }
}

export class MockTranslateService {
  get() {
    /** */
  }
}
