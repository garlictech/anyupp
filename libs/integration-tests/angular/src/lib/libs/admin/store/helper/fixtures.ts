export const testFixtures = [
  {
    id: 'test_fixture_id_1',
    name: 'Text fixture 1',
  },
  {
    id: 'test_fixture_id_2',
    name: 'Text fixture 2',
  },
  {
    id: 'test_fixture_id_3',
    name: 'Text fixture 3',
  },
];
