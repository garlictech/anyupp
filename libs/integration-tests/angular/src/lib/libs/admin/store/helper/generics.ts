/* eslint-disable @typescript-eslint/no-explicit-any */

import { of, range } from 'rxjs';
import { delay, last, switchMap, take, tap } from 'rxjs/operators';

export const emptyListTestLogic = (service: any, done: jest.DoneCallback) => {
  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The initial list is empty
        expect(items).toEqual([]);
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const filteredListTestLogic = (
  service: any,
  dataFixtures: any[],
  filter: any,
  done: jest.DoneCallback,
) => {
  const sdkSpy = jest.spyOn(service.apiConf, 'add');

  range(0, dataFixtures.length)
    .pipe(
      tap(idx => {
        sdkSpy.mockImplementation(
          jest.fn().mockReturnValue(of(dataFixtures[idx])),
        );
        service.setFilter(filter);
      }),
      switchMap(idx => service.add$(dataFixtures[idx])),
      last(),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // Full list
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.filteredEntities$),
      take(1),
      tap(items => {
        // Filtered list
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const addEntityToListTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  const sdkSpy = jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service
    .add$(dataFixture)
    .pipe(
      tap(() => {
        // "Add" method has been called
        expect((sdkSpy.mock.calls[0][0] as any).input).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // The list has been updated
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const updateEntityListTestLogic = (
  service: any,
  dataFixture: any,
  updateFixture: any,
  done: jest.DoneCallback,
) => {
  const sdkCreateSpy = jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkUpdateSpy = jest
    .spyOn(service.apiConf, 'update')
    .mockImplementation(jest.fn().mockReturnValue(of(updateFixture)));

  service
    .add$(dataFixture)
    .pipe(
      tap(() => {
        // "Add" method has been called
        expect((sdkCreateSpy.mock.calls[0][0] as any).input).toMatchSnapshot();
      }),
      switchMap(() => service.update$(updateFixture)),
      tap(() => {
        // "Update" method has been called
        expect((sdkUpdateSpy.mock.calls[0][0] as any).input).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // The list has been updated
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const patchFilterTestLogic = (
  service: any,
  initialFilter: Record<string, unknown>,
  newFilter: Record<string, unknown>,
  done: jest.DoneCallback,
) => {
  of('test')
    .pipe(
      tap(() => {
        // Initial filter
        service.setFilter(initialFilter);
      }),
      switchMap(() => service.filter$),
      take(1),
      tap(filters => {
        // Filter has been updated
        expect(filters).toMatchSnapshot();
        service.patchFilter(newFilter);
      }),
      switchMap(() => service.filter$),
      take(1),
      tap(filters => {
        // Filter has been updated
        expect(filters).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromApiTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  const sdkSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service
    .getByKey$(dataFixture.id)
    .pipe(
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      tap(() => {
        // "Get" method has been called
        expect((sdkSpy.mock.calls[0][0] as any).id).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // New item stored in the entity list
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromCacheTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkGetSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.add$(dataFixture)),
      tap(items => {
        // The new item has been added
        expect(items).toMatchSnapshot();
      }),
      delay(2000),
      switchMap(() => service.getByKey$(dataFixture.id)),
      tap(item => {
        // GET query has not been called
        expect(sdkGetSpy).not.toHaveBeenCalled();

        // Got the item from the cache
        expect(item).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromCacheForceApiTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkGetSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.add$(dataFixture)),
      tap(items => {
        // The new item has been added
        expect(items).toMatchSnapshot();
      }),
      delay(2000),
      switchMap(() => service.getByKey$(dataFixture.id, true, true)),
      tap(item => {
        // "Get" method has been called
        expect((sdkGetSpy.mock.calls[0][0] as any).id).toMatchSnapshot();

        // Got the item from the cache
        expect(item).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};

export const getByKeyFromApiAndNotSaveToListTestLogic = (
  service: any,
  dataFixture: any,
  done: jest.DoneCallback,
) => {
  jest
    .spyOn(service.apiConf, 'add')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  const sdkGetSpy = jest
    .spyOn(service.apiConf, 'get')
    .mockImplementation(jest.fn().mockReturnValue(of(dataFixture)));

  service.entities$
    .pipe(
      take(1),
      tap(items => {
        // The entity list is empty
        expect(items).toMatchSnapshot();
      }),
      switchMap(() => service.getByKey$(dataFixture.id, false, true)),
      tap(item => {
        // "Get" method has been called
        expect((sdkGetSpy.mock.calls[0][0] as any).id).toMatchSnapshot();

        // Got the item from the cache
        expect(item).toMatchSnapshot();
      }),
      switchMap(() => service.entities$),
      take(1),
      tap(items => {
        // New item NOT stored in the entity list
        expect(items).toMatchSnapshot();
      }),
    )
    .subscribe(() => {
      done();
    });
};
