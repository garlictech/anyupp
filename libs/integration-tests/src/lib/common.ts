import { devConfig as config } from '@bgap/shared/config';
export const testAdminUsername = 'test@anyupp.com';
export const testAdminUserPassword = 'Testtesttest12_';

const { GraphqlApiKey, GraphqlApiUrl } = config;
export { GraphqlApiUrl, GraphqlApiKey };
