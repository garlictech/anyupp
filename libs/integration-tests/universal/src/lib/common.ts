import { config } from '@bgap/shared/config';
export const testAdminUsername = 'test@anyupp.com';
export const testAdminUserPassword = 'Testtesttest12_';
import Amplify from 'aws-amplify';
import { awsConfig } from '@bgap/admin/amplify-api';

const { GraphqlApiKey, GraphqlApiUrl } = config;
export { GraphqlApiUrl, GraphqlApiKey };

export const configureAmplify = () => {
  Amplify.configure({
    ...awsConfig,
    // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });
};
