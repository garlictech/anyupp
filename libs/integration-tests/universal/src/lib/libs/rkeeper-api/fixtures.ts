import * as CrudApi from '@bgap/crud-gql/api';
import {
  Dish,
  RKeeperBusinessEntityInfo,
  RkeeperFixtures,
} from '@bgap/rkeeper-api';
import { RequiredId } from '@bgap/shared/types';
import {
  unitFixture,
  productFixture,
  chainFixture,
  groupFixture,
  rkeeperEndpoint,
  yellowRestaurantId,
  yellowRkeeperUsername,
  yellowRkeeperPassword,
} from '@bgap/shared/fixtures';

const testIdPrefix = `rkeeper-cf0d1110-a2ce-45cf-aa69-6782bbc44cad`;

export const rawData = {
  data: {
    dishes: [
      RkeeperFixtures.dish,
      RkeeperFixtures.dish2,
      RkeeperFixtures.duplicatedDish,
      RkeeperFixtures.badDish,
      RkeeperFixtures.inactiveDish,
      RkeeperFixtures.dishWithModifier,
      RkeeperFixtures.dishWithSameModifier,
    ],
    modifiers: [RkeeperFixtures.modifier, RkeeperFixtures.unusedModifier],
  },
};

export const rkeeperProductGuid = 'RKEEPERGUID';

const unitId = `${testIdPrefix}-unit`;
const chainId = `${testIdPrefix}-chain`;
const groupId = `${testIdPrefix}-group`;

export const rkeeperUnit: RequiredId<CrudApi.CreateUnitInput> = {
  ...unitFixture.createRkeeperUnit,
  id: unitId,
  groupId,
  chainId,
  externalId: 'EXTERNAL-RESTAURANT-ID',
};

export const rkeeperUnitProduct: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct`,
  externalId: rkeeperProductGuid,
};

export const rkeeperUnitProduct2: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct2`,
  externalId: rkeeperProductGuid + '2',
};

export const processedDish: Dish = {
  price: 50000,
  active: true,
  id: 9991000114,
  guid: '999e3ab3-86a0-48d9-a9a9-f4e0c9fbce68',
  name: 'Pr√≥ba ital',
};

export const businessEntity: RKeeperBusinessEntityInfo = {
  chainId,
  groupId,
  unitId,
};

export const createChain: RequiredId<CrudApi.CreateChainInput> = {
  ...chainFixture.chainBase,
  id: chainId,
};

export const createGroup: RequiredId<CrudApi.CreateGroupInput> = {
  ...groupFixture.groupBase,
  id: groupId,
  chainId,
};

export const rkeeperOrder = {
  objectid: yellowRestaurantId,
  remoteId: yellowRestaurantId,
  order_type: 1,
  pay_type: 0,
  pay_online_type: 0,
  delivery_time: '2122-02-03T12:37:32',
  client: {
    phone: null,
    email: null,
    ln: 'Testln',
    fn: 'Testfn',
  },
  order: [
    {
      id: '1000114',
      type: 'd',
      qnt: 1000,
    },
  ],
};

export const yellowUnit = {
  ...rkeeperUnit,
  externalId: yellowRestaurantId,
  pos: {
    ...rkeeperUnit.pos,
    rkeeper: {
      ...(rkeeperUnit?.pos?.rkeeper || {}),
      rkeeperUsername: yellowRkeeperUsername,
      rkeeperPassword: yellowRkeeperPassword,
      endpointUri: rkeeperEndpoint,
    } as CrudApi.RKeeper,
  },
} as CrudApi.Unit;

export const orderInput: CrudApi.CreateOrderInput = {
  userId: 'USER_ID',
  unitId: yellowUnit.id,
  items: [
    {
      quantity: 5,
      productId: 'PRODUCT ID',
      statusLog: [
        {
          userId: 'USER_ID',
          status: CrudApi.OrderStatus.none,
          ts: 1627909024677,
        },
      ],
      sumPriceShown: {
        taxSum: 316.98,
        currency: 'HUF',
        tax: 27,
        priceSum: 1491,
        pricePerUnit: 298.2,
      },
      productName: {
        hu: 'PRODUCT NAME HU',
      },
      priceShown: {
        taxSum: 318.9,
        currency: 'HUF',
        tax: 27,
        priceSum: 1500,
        pricePerUnit: 300,
      },
      variantId: 'VARIANT ID',
      variantName: {
        en: 'VARIANT NAME EN',
      },
    },
  ],
  sumPriceShown: {
    taxSum: 633.96,
    currency: 'HUF',
    tax: 27,
    priceSum: 2982,
    pricePerUnit: 298.2,
  },
  place: {
    table: '01',
    seat: '01',
  },
  orderMode: CrudApi.OrderMode.instant,
  servingMode: CrudApi.ServingMode.inplace,
  archived: false,
  paymentMode: {
    method: CrudApi.PaymentMethod.cash,
    type: CrudApi.PaymentType.cash,
  },
};

export {
  yellowRestaurantId,
  yellowRkeeperPassword,
  yellowRkeeperUsername,
  rkeeperEndpoint,
};
