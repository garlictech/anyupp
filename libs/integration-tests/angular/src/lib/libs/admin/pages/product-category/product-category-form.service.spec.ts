import { EMPTY, of } from 'rxjs';
import { catchError, switchMap, tap } from 'rxjs/operators';

import { TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { ProductCategoryFormService } from '@bgap/admin/pages/product-categories';
import { CrudSdkService } from '@bgap/admin/shared/data-access/sdk';
import * as CrudApi from '@bgap/crud-gql/api';
import { productCategoryFixture, testIdPrefix } from '@bgap/shared/fixtures';
import { UpsertResponse } from '@bgap/shared/types';
import { StoreModule } from '@ngrx/store';

import { signInToCognito, signOutFromCognito } from '../../shared/helper';
import { EffectsModule } from '@ngrx/effects';
import {
  EntityCollectionServiceElementsFactory,
  EntityDataModule,
  EntityDispatcherFactory,
} from '@ngrx/data';
import { entityConfig } from '@bgap/admin/shared/data-access/ngrx-data';
import { HttpClientModule } from '@angular/common/http';
import { AdminSharedLoggedUserModule } from '@bgap/admin/store/logged-user';

describe('ProductCategoryFormService', () => {
  const productCategoryId = `${testIdPrefix}ADMIN_PRODUCT_CATEGORY_IT_PRODUCT_CATEGORY_ID_01`;

  let service: ProductCategoryFormService;
  let crudSdk: CrudSdkService;

  const cleanup = () =>
    crudSdk.sdk.DeleteProductCategory({
      input: { id: productCategoryId },
    });

  beforeAll(async () => {
    await signInToCognito();
  }, 60000);

  afterAll(async () => {
    await signOutFromCognito();
  }, 60000);

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        StoreModule.forRoot({}),
        EffectsModule.forRoot([]),
        EntityDataModule.forRoot(entityConfig),
        HttpClientModule,
        AdminSharedLoggedUserModule,
      ],
      providers: [
        EntityCollectionServiceElementsFactory,
        EntityDispatcherFactory,
      ],
    });

    service = TestBed.inject(ProductCategoryFormService);
    crudSdk = TestBed.inject(CrudSdkService);
  }, 60000);

  it('createGroupFormGroup should create form group', () => {
    expect(service.createProductCategoryFormGroup().value).toMatchSnapshot();
  });

  it('saveForm$ should call createProductCategory$ method when id is not specified', done => {
    const createSpy = jest
      .spyOn(service, 'createProductCategory$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'insert' }));

    service.saveForm$(productCategoryFixture.productCategoryBase).subscribe();

    expect(createSpy).toHaveBeenCalledWith(
      productCategoryFixture.productCategoryBase,
    );

    done();
  }, 60000);

  it('saveForm$ should call updateProductCategory$ method when id is specified', done => {
    const updateSpy = jest
      .spyOn(service, 'updateProductCategory$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveForm$(productCategoryFixture.productCategoryBase, productCategoryId)
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith({
      ...productCategoryFixture.productCategoryBase,
      id: productCategoryId,
    });

    done();
  }, 60000);

  it('createProductCategory$ should create product category', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createProductCategory$({
            ...productCategoryFixture.productCategoryBase,
            id: productCategoryId,
          }),
        ),
        tap(saveResponse => {
          expect(saveResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 60000);

  it('updateProductCategory$ should update product category', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createProductCategory$({
            ...productCategoryFixture.productCategoryBase,
            id: productCategoryId,
          }),
        ),
        catchError(() => cleanup()),
        switchMap(saveResponse =>
          (<UpsertResponse<CrudApi.ProductCategory>>saveResponse).data.id
            ? service.updateProductCategory$({
                ...productCategoryFixture.productCategoryBase,
                id: productCategoryId,
                image: 'MODDED IMAGE',
                position: 33,
              })
            : EMPTY,
        ),
        catchError(() => cleanup()),
        tap(updateResponse => {
          expect(updateResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 60000);
});
