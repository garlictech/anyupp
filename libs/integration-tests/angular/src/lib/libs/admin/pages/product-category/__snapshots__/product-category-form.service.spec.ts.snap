// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategoryFormService createGroupFormGroup should create form group 1`] = `
Object {
  "description": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
  "image": "",
  "name": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
}
`;

exports[`ProductCategoryFormService createProductCategory$ should create product category 1`] = `
Object {
  "data": Object {
    "chainId": "seeded_chain_1_id",
    "createdAt": Any<String>,
    "description": Object {
      "de": null,
      "en": "DESCRIPTION",
      "hu": null,
    },
    "id": "test_ADMIN_PRODUCT_CATEGORY_IT_PRODUCT_CATEGORY_ID_01",
    "image": "IMAGE",
    "name": Object {
      "de": null,
      "en": "PRODUCT_CATEGORY",
      "hu": null,
    },
    "parentId": null,
    "position": 0,
    "updatedAt": Any<String>,
  },
  "type": "insert",
}
`;

exports[`ProductCategoryFormService updateProductCategory$ should update product category 1`] = `
Object {
  "data": Object {
    "chainId": "seeded_chain_1_id",
    "createdAt": Any<String>,
    "description": Object {
      "de": null,
      "en": "DESCRIPTION",
      "hu": null,
    },
    "id": "test_ADMIN_PRODUCT_CATEGORY_IT_PRODUCT_CATEGORY_ID_01",
    "image": "MODDED IMAGE",
    "name": Object {
      "de": null,
      "en": "PRODUCT_CATEGORY",
      "hu": null,
    },
    "parentId": null,
    "position": 33,
    "updatedAt": Any<String>,
  },
  "type": "update",
}
`;
